bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java

```diff
@@ -615,4 +623,24 @@ public void stackView(String id, String refId, boolean visible) {
 		}
 	}
 
+	@Override
+	public void setEditorOnboardingText(String text) {
+		perspModel.getTags().add(EDITOR_ONBOARDING_TEXT + text);
+	}
+
+	@Override
+	public void setEditorOnboardingImage(String iconUri) {
+		perspModel.getTags().add(EDITOR_ONBOARDING_IMAGE + iconUri);
+	}
+
+	@Override
+	public void addEditorOnboardingCommand(String commandId) {
+		Predicate<MKeyBinding> commandWithEqualId = b -> b.getCommand().getElementId().equals(commandId);
+		Function<MKeyBinding, String> toCommandText = b -> b.getCommand().getCommandName() + ": " + b.getKeySequence(); //$NON-NLS-1$
```

Comment:
Instead of
b.getCommand().getCommandName() + ": " + b.getKeySequence()
you should use something like:
```suggestion
		b.getCommand().getCommandName() + ": " //$NON-NLS-1$
						+ org.eclipse.jface.bindings.keys.KeySequence.getInstance(b.getKeySequence()).format();
```

to get it formatted correctly on macOS